### PR TITLE
change restore dir

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#2987

File: src/docfx/restore/Restore.cs [hunk 1]
```diff
@@ -52,6 +52,22 @@ public static (string restoreDir, string url, string refSpec) GetGitRestoreInfo(
...
         }
 
+        /// <summary>
+        /// Get the restore root dir, default is the user proflie dir.
+        /// User can set the DOCFX_APPDATA_PATH environment to change the root
+        /// </summary>
+        public static string GetRestoreRoot()
+        {
+            // TODO: document this environment variable and show it in welcome message
+            var docfxAppData = Environment.GetEnvironmentVariable("DOCFX_APPDATA_PATH");
```
Comment (yufeih) [inline src/docfx/restore/Restore.cs]: this also affects lock, user profile cache. add a static `AppDataPath` to manage them all.

File: src/docfx/config/AppData.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        private static readonly string s_appDataPath = GetAppDataPath();
+
+        /// <summary>
+        /// Get the restore root dir, default is the user proflie dir.
+        /// User can set the DOCFX_APPDATA_PATH environment to change the root
+        /// </summary>
+        private static string GetAppDataPath()
+        {
+            // TODO: document this environment variable and show it in welcome message
```
Comment (yufeih) [inline src/docfx/config/AppData.cs]: we could document this, but don't show it in welcome :)

File: src/docfx/config/AppData.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            var docfxAppData = Environment.GetEnvironmentVariable("DOCFX_APPDATA_PATH");
+            if (!string.IsNullOrEmpty(docfxAppData))
+            {
+                docfxAppData = Path.GetFullPath(docfxAppData);
+            }
+
+            return docfxAppData ?? Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
```
Comment (yufeih) [inline src/docfx/config/AppData.cs]: seem like empty string leaks through, how about
```csharp
string.IsNullOrEmpty(docfxAppData) ? UserProfile : GetFullPath(appdata)
```

Comment (OsmondJiang) [reply src/docfx/config/AppData.cs]: unless use set a environment value is ""...

**Pull request merged**